pkg/operator/ceph/cluster/controller.go

```diff
@@ -271,10 +271,24 @@ func (c *ClusterController) onAdd(obj interface{}) {
 		logger.Warningf("mon count is even (given: %d), should be uneven, continuing", cluster.Spec.Mon.Count)
 	}
 
-	cephVersion, err := cluster.detectCephVersion(cluster.Spec.CephVersion.Image, 15*time.Minute)
-	if err != nil {
-		logger.Errorf("unknown ceph major version. %+v", err)
-		return
+	retryCount := 0
+	retryInterval := time.Duration(versionDetectionRetryInternval) * time.Second
+	// initialize a dummy cephVersion
+	cephVersion := &version.CephVersion{}
+	for {
+		cephVersion, err = cluster.detectCephVersion(cluster.Spec.CephVersion.Image, 15*time.Minute)
```

Comment:
What if we just moved the call to `detectCephVersion()` inside the `wait.Poll()` on line 302 below? Then we will get the retry for free. We just wouldn't want to retry if it's an unsupported version according to line 296.